src/cps/loader.cpp

```diff
@@ -206,23 +207,49 @@ namespace cps::loader {
+                auto const compile_flags = CPS_TRY(get_lang_values(comp, name, "compile_flags"));
+                auto const includes = CPS_TRY(get_lang_values(comp, name, "includes"));
+                auto const defines = CPS_TRY(get_defines(comp, name, "defines"));
+                auto const link_flags = CPS_TRY(get_optional<std::vector<std::string>>(comp, name, "link_flags"))
+                                            .value_or(std::vector<std::string>{});
+                auto const link_libraries =
                     CPS_TRY(get_optional<std::vector<std::string>>(comp, name, "link_libraries"))
-                        .value_or(std::vector<std::string>{}),
-                    // TODO: this is required if the type != interface
-                    CPS_TRY(get_optional<std::string>(comp, name, "location")),
-                    // XXX: https://github.com/cps-org/cps/issues/34
-                    CPS_TRY(get_optional<std::string>(comp, name, "link_location")),
-                    CPS_TRY(get_optional<std::vector<std::string>>(comp, name, "requires"))
-                        .value_or(std::vector<std::string>{})};
+                        .value_or(std::vector<std::string>{});
+                auto const location = CPS_TRY(get_optional<std::string>(comp, name, "location"));
+                auto const link_location = CPS_TRY(get_optional<std::string>(comp, name, "link_location"));
+                auto const require = CPS_TRY(get_optional<std::vector<std::string>>(comp, name, "requires"))
+                                         .value_or(std::vector<std::string>{});
+
+                if (type == Type::unknown) {
+                    continue;
+                }
+                if (type == Type::archive && !location.has_value()) {
+                    return tl::make_unexpected(
+                        fmt::format("component `{}` of type `archive` missing required key `location`", key));
+                }
+                // TODO: Validate link_location, see https://github.com/cps-org/cps/issues/34
+
+                components[key] = Component{
+                    .type = std::move(type),
+                    .compile_flags = std::move(compile_flags),
+                    .includes = std::move(includes),
+                    .defines = std::move(defines),
+                    .link_flags = std::move(link_flags),
+                    .link_libraries = std::move(link_libraries),
+                    .location = std::move(location),
+                    .link_location = std::move(link_location),
+                    .require = std::move(require),
+                };
+            }
+
+            if (components.empty()) {
+                return tl::unexpected(fmt::format("Components field of `{}` is empty, but must "
+                                                  "have at least one component",
+                                                  parent_name));
```

Comment:
```suggestion
                return tl::unexpected(fmt::format("`{}` must have at least one component",
                                                  parent_name));
```
...note that I haven't checked if clang-tidy is happy with this formatting.